File: algorithm笔记/快速排序/quick.js
function quickSort(arr) {
	if (arr.length <= 1) return arr;

	let baseIndex = Math.floor(arr.length / 2);
	let base = arr.splice(baseIndex, 1)[0];

	let left = [];
	let right = [];

	for (let i = 0; i < arr.length; i++) {
		if (arr[i] < base) {
			left.push(arr[i]);
		} else {
			right.push(arr[i]);
		}
	}

	return quickSort(left).concat([base], quickSort(right))
}

File: z____stash/algorithm笔记/快速排序/quick.py
import math
def quickSort(arr):
	length = len(arr)
	if length <= 1:
		return arr
	baseIndex = math.floor(length / 2)
	base = arr[baseIndex : baseIndex + 1][0]
	left = []
	right = []
	for index in range(length):
		if arr[index] < base:
			left.append(arr[index])
		else:
			right.append(arr[index])
	return quickSort(left) + [base] + quickSort(right)